### DISCUSSION TITLE
Encoding for Arabic and handling request.post

### DISCUSSION THREAD

**Discussion by elsayed-issa** [fastapi/fastapi#6232] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from __future__ import annotations
from typing import  Text
from pydantic import BaseModel
from fastapi import FastAPI
import uvicorn

app = FastAPI()

class Result(BaseModel):
    SingularForm: Text
    PluralForm: Text
    SingularPattern: Text
    PluralPattern: Text

def search(word: Text):
    res = {
          "SingularForm": "خبر",
          "PluralForm": "أَخْبار",
          "SingularPattern": "فَعَل",
          "PluralPattern": "أَفْعال"
        }
    if word:
        return res

class ShowResult(Result):
    class Config():
        orm_mode = True

@app.get('/')
def home():
    return {'message': 'System is working!'}

@app.post("/query", response_model=ShowResult)
def post_query(query: Text):
    return get_query(query)

@app.get('/query', response_model=ShowResult)
def get_query(word: Text):
    query = search(word)
    return query
    
if __name__ == '__main__':
    uvicorn.run("test:app", host="0.0.0.0", port=8000, reload=True)


def predict(query: Text):
    headers = {
        'accept': 'application/json',
        'Content-Type': 'application/json; charset=UTF-8'
            }

    data = {'query':query}

    response = requests.post("http://0.0.0.0:8000/query", headers=headers, data= json.dumps(data))
    #results = response.json()
    return response.content

print(predict('خبر')
```

### Description

Hi, I am building a system using FastAPI to search for an Arabic verb using the `search` function shown in the code snippet. This function returns the Pydantic class `Result`. When I test this code snippet in FastAPI docs, both `get_query` and `post_query` functions work perfect and I get the results. However, when I use the function `predict`, I always get the following error 
`{"detail":[{"loc":["query","query"],"msg":"field required","type":"value_error.missing"}]}`

Before that error, I had an encoding error, so I added `charset=UTF-8` to `headers` and it disappeared.

I hope it is clear and thank you for your help.

### Operating System

macOS

### FastAPI Version

0.78.0

### Python Version

Python 3.8.9

**Comment by sbusch-mobivention** [comment, marked as answer]
This has nothing to do with the language and only with how FastAPI treats different types of parameters. If you use a singular string parameter in your route, FastAPI will expect that to be a query param, but in predict you send it as a json body, so it will be missing.

There are 2 options to fix this.
Either send the query as a query param with requests:
```
data = {"query": query}

response = requests.post("http://0.0.0.0:8000/query", headers=headers, params=data)
```

Or change your route so it accepts a json object:
```
class QueryRequest(BaseModel):
    query: Text


@app.post("/query", response_model=ShowResult)
def post_query(req: QueryRequest):
    return get_query(req.query)
```

**Comment by elsayed-issa** [comment]
Thank you so much, @sbusch-mobivention. I appreciate it.
